lib/cylc/batch_sys_handlers/lsf.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2015 NIWA
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"IBM Platform LSF bsub job submission"
+
+import re
+
+
+class LSFHandler(object):
+    "IBM Platform LSF bsub job submission"
+
+    DIRECTIVE_PREFIX = "#BSUB "
+    KILL_CMD_TMPL = "bkill '%(job_id)s'"
+    POLL_CMD_TMPL = "bjobs -noheader '%(job_id)s'"
+    REC_ID_FROM_SUBMIT_OUT = re.compile(r"^Job <(?P<id>\d+)>")
+    SUBMIT_CMD_TMPL = "bsub"
+    SUBMIT_CMD_STDIN_EXEC_TMPL = "cat '%(job)s'"
```

Comment:
We can make an assumption like this:

```python
if "%(job)s" in batch_sys.SUBMIT_CMD_TMPL:
    # Normal
elif hasattr(batch_sys, "SUBMIT_CMD_STDIN_TMPL"):
    # Like "at"
else:
    # Do "Popen" with "stdin=open(job_file_path)"
```

Or perhaps have a simple attribute like `SUBMIT_CMD_STDIN_JOB_FILE = True`:

```python
if getattr(batch_sys, "SUBMIT_CMD_STDIN_JOB_FILE", False):
    # Do "Popen" with "stdin=open(job_file_path)"
else:
    # Normal
```